Core/src/Testing/System/SystemTestCase.php

```diff
@@ -97,8 +98,14 @@ public static function randId()
      */
     public static function createBucket(StorageClient $client, $bucketName, array $options = [])
     {
-        $backoff = new ExponentialBackoff(8);
+        $backoff = new ExponentialBackoff(8, function ($ex) {
+            if ($ex instanceof BadRequestException) {
+                // no retry for malformed requests
+                return $ex->getCode() !== 400;
+            }
 
+            return true;
```

Comment:
@dwsupplee should I do something like below here? I am not sure whether I should skip retry for all 4XX & 5XX failures _OR_ just 400. 
Code pointer ([link](https://github.com/googleapis/google-cloud-php/blob/a707e9dc29d9f4b1fa02813573505e9426c918f4/Core/src/RequestWrapper.php#L344-L347)) where the failures are parsed.

```suggestion
            return !($ex instanceof BadRequestException);
```

Reply:
This approach looks good to me and fits our use case well.